network/src/traits.rs

```diff
@@ -104,3 +105,80 @@ pub trait ReliableNetwork: BaseNetwork {
         handlers
     }
 }
+
+#[async_trait]
+pub trait UnreliableNetwork2<Message: Clone + Send + Sync> {
+    async fn unreliable_send(&mut self, peer: PublicKey, message: &Message) -> JoinHandle<()>;
+
+    /// Broadcasts a message to all `peers` passed as an argument.
+    /// The attempts to send individual messages are best effort and will not be retried.
+    async fn unreliable_broadcast(
+        &mut self,
+        peers: Vec<PublicKey>,
+        message: &Message,
+    ) -> Vec<JoinHandle<()>> {
+        let mut handlers = Vec::new();
+        for peer in peers {
+            let handle = { self.unreliable_send(peer, message).await };
+            handlers.push(handle);
+        }
+        handlers
+    }
+}
+
+pub trait Lucky {
+    fn rng(&mut self) -> &mut SmallRng;
+}
+
+#[async_trait]
+pub trait LuckyNetwork2<Message> {
+    /// Pick a few addresses at random (specified by `nodes`) and try (best-effort) to send the
+    /// message only to them. This is useful to pick nodes with whom to sync.
+    async fn lucky_broadcast(
+        &mut self,
+        mut peers: Vec<PublicKey>,
+        message: &Message,
+        nodes: usize,
```

Comment:
I think a slightly more verbose name like `num_nodes` would be clearer here (unless this is already following an established convention?)

Or maybe even `num_peers` or `num_targets`? Curious why the word "nodes" - does that have a specific meaning in this context?

Reply:
each entity is a "node" in the network, but yeah i'll change to num_nodes.